specifications/Disruptor/Disruptor_SPMC.tla

```diff
@@ -132,14 +134,18 @@ Next ==
   \/ \E r \in Readers : EndRead(r)
 
 Fairness ==
-  /\ \A w \in Writers : WF_vars(BeginWrite(w))
-  /\ \A w \in Writers : WF_vars(EndWrite(w))
```

Comment:
I find it surprising that a started write doesn't have to be ended.  Isn't the `EndWrite` action necessary so that another writer can eventually claim that slot?

Reply:
This is the single producer case so there's just one.
Also, I think it actually makes sense for a producer to stutter after it has begun a write. It just means the consumers will never read that event. But on the other hand, the consumers will read the event if a producer finishes a write (as described in the liveliness property.)